debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java

```diff
@@ -544,4 +546,42 @@ public NonRelationalTableException(String message) {
             super(message);
         }
     }
+
+    public static class OracleConnectionConfiguration {
+        protected static final String JDBC_PROPERTY_JSON_DEFAULT_GET_OBJECT_TYPE = "oracle.jdbc.jsonDefaultGetObjectType";
+
+        private final JdbcConfiguration jdbcConfig;
+        private final Configuration config;
+        private final ConnectionFactory factory;
+
+        public OracleConnectionConfiguration(JdbcConfiguration config) {
+            // Set up the JDBC connection without actually connecting, with extra Oracle-specific properties
+            // to handle JSON columns in Oracle 23+ as per
+            // https://docs.oracle.com/en//database/oracle/oracle-database/23/jajdb/oracle/jdbc/OracleConnection.html#CONNECTION_PROPERTY_JSON_DEFAULT_GET_OBJECT_TYPE
+            this.config = config;
+
+            final Configuration dbConfig = config
+                    .merge(config.subset(DRIVER_CONFIG_PREFIX, true));
+
+            final Configuration.Builder jdbcConfigBuilder = dbConfig
+                    .edit()
+                    .with(JDBC_PROPERTY_JSON_DEFAULT_GET_OBJECT_TYPE, "java.lang.String");
```

Comment:
If a user specifies this connector configuration argument in the connector configuration, it would seem to me that using `edit().with(...)` would override the user-provided property.  I believe in this context, we likely want to use `edit().withDefault(...)` instead.